zerver/tests/test_outgoing_webhook_system.py

```diff
@@ -316,6 +317,45 @@ def test_pm_to_outgoing_webhook_bot(self, mock_requests_request: mock.Mock) -> N
             Recipient.PERSONAL,
         )
 
+    def test_pm_to_outgoing_webhook_bot_for_407_error_code(self) -> None:
+        bot_owner = self.example_user("othello")
+        bot = self.create_outgoing_bot(bot_owner)
+        sender = self.example_user("hamlet")
+        realm = get_realm("zulip")
+
+        response = ResponseMock(407)
+        expect_407 = mock.patch("requests.request", return_value=response)
+        expect_fail = mock.patch("zerver.lib.outgoing_webhook.fail_with_message")
+
+        with expect_407, expect_fail as mock_fail, self.assertLogs(level="WARNING"):
+            message_id = self.send_personal_message(sender, bot, content="foo")
+
+            # create message dict to get the message url
+            message = {
+                "display_recipient": [{"id": bot.id}, {"id": sender.id}],
+                "stream_id": 999,
+                TOPIC_NAME: "Foo",
+                "id": message_id,
+                "type": "",
+            }
+            message_url = near_message_url(realm, message)
+
+            last_message = self.get_last_message()
+            self.assertEqual(
+                last_message.content,
+                f"[A message]({message_url}) to your @**{bot.full_name}** triggered an outgoing webhook.\nAccess to the outgoing webhook's network is not allowed.",
```

Comment:
I'd split this string across 2 lines at the newline character, if convenient; would be more readable.